src/NanopassSharp/Transformations/TransformationExtensions.cs

```diff
@@ -0,0 +1,15 @@
+﻿using NanopassSharp.Descriptions;
+
+namespace NanopassSharp.Transformations;
+
+public static class TransformationExtensions
+{
+    /// <summary>
+    /// Creates an <see cref="ITransformationDescription"/> from a pattern and a transformation.
+    /// </summary>
+    /// <param name="transformation">The transformation.</param>
+    /// <param name="pattern">The pattern.</param>
+    /// <returns>A new transformation description.</returns>
+    public static ITransformationDescription WithPattern(this ITransformation transformation, ITransformationPattern? pattern) =>
```

Comment:
I personally don't see this being too useful other than having a fluent API to build up a transformation desc just for the sake of it.
I'll likely build up both and then construct the desc. straight away.